### PR TITLE
Initialize custom properties with default values on non-interactive document creations

### PR DESCRIPTION
**Pull request by lukasgraf**
This initializes custom properties with default values for

- Documents and mails **uploaded with TUS**
- Documents and mails **uploaded via QuickUpload**
- Mails received via **inbound mail**
- Documents **extracted from mail attachments**

For [CA-2482](https://4teamwork.atlassian.net/browse/CA-2482)

## Checklist

- [ ] Changelog entry (*not necessary - fix for behavior that was introduced in the same release*)
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

- New functionality:
  - [x] for `document` also works for `mail`

### CODE REVIEW

File: opengever/propertysheets/tests/test_creation_defaults.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            )
+            .with_field(
+                'textline', u'notrequired', u'Optional field with default', u'',
+                required=False,
+                default=u'Not required, still has default'
+            )
+            .with_field(
+                'multiple_choice', u'languages', u'Languages', u'',
+                required=True, values=[u'de', u'fr', u'en'],
+                default={u'de', u'en'},
```
Comment (deiferni) [inline opengever/propertysheets/tests/test_creation_defaults.py:58]: This set instance here could be potentially referenced when setting it on an object leading to shared state/defaults. Is there some functionality somewhere guarding against this?

File: opengever/propertysheets/creation_defaults.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    storage = PropertySheetSchemaStorage()
+    for slot_name in propsheet_field.valid_assignment_slots_factory():
+        slot_values = {}
+        definition = storage.query(slot_name)
+
+        if definition is not None:
+            for (field_name, field) in definition.get_fields():
+                if field.default is not None:
+                    slot_values[field.__name__] = field.default
```
Comment (deiferni) [inline opengever/propertysheets/creation_defaults.py:41]: Ah here it is … you might have to guard against mutable defaults and create a copy of sets/lists/…. See previous comment. Unless of course the field already takes care of such things while accessing the default.

**Review by deiferni — APPROVED** [review]
LGTM 👍 

I've discovered one potential issue where the default (mutable) might be shared between objects when setting it from the fields default. Not sure what happens under the hood there when retrieving the default but best have a close look before we ship.

File: opengever/propertysheets/creation_defaults.py (hunk 2 above)
Comment (lukasgraf) [reply opengever/propertysheets/creation_defaults.py]: Great catch! 🏹 
It was indeed an issue. 💥  Added a failing test and a fix.

**Comment by lukasgraf** [comment]
@deiferni thanks for the review! I addressed the mutable kwarg issue, and also decided to make the `Location` header parsing in TUS upload more robust (last 3 commits). Would you mind having another quick look?

**Pull request merged**